parser/parser_test.go

```diff
@@ -384,6 +301,112 @@ AAAAAAAAAAAA5gm5Mg==
 	s.RunTest(c, table)
 }
 
+func (s *testParserSuite) TestDBAStmt(c *C) {
+	defer testleak.AfterTest(c)()
+	table := []testCase{
+		// For SHOW statement
+		{"SHOW VARIABLES LIKE 'character_set_results'", true},
+		{"SHOW GLOBAL VARIABLES LIKE 'character_set_results'", true},
+		{"SHOW SESSION VARIABLES LIKE 'character_set_results'", true},
+		{"SHOW VARIABLES", true},
+		{"SHOW GLOBAL VARIABLES", true},
+		{"SHOW GLOBAL VARIABLES WHERE Variable_name = 'autocommit'", true},
+		{"SHOW STATUS", true},
+		{"SHOW GLOBAL STATUS", true},
+		{"SHOW SESSION STATUS", true},
+		{"SHOW STATUS LIKE 'Up%'", true},
+		{"SHOW STATUS WHERE Variable_name LIKE 'Up%'", true},
+		{`SHOW FULL TABLES FROM icar_qa LIKE play_evolutions`, true},
+		{`SHOW FULL TABLES WHERE Table_Type != 'VIEW'`, true},
+		{`SHOW GRANTS`, true},
+		{`SHOW GRANTS FOR 'test'@'localhost'`, true},
+		{`SHOW COLUMNS FROM City;`, true},
+		{`SHOW FIELDS FROM City;`, true},
+		{`SHOW TRIGGERS LIKE 't'`, true},
+		{`SHOW DATABASES LIKE 'test2'`, true},
+		{`SHOW PROCEDURE STATUS WHERE Db='test'`, true},
+		{`SHOW INDEX FROM t;`, true},
+		{`SHOW KEYS FROM t;`, true},
+		// For show character set
+		{"show character set;", true},
+		// For show collation
+		{"show collation", true},
+		{"show collation like 'utf8%'", true},
+		{"show collation where Charset = 'utf8' and Collation = 'utf8_bin'", true},
+		// For show full columns
+		{"show columns in t;", true},
+		{"show full columns in t;", true},
+		// For show create table
+		{"show create table test.t", true},
+		{"show create table t", true},
+
+		// set
+		// user defined
+		{"SET @a = 1", true},
```

Comment:
SET *is* DBA statement, you can find it here:

https://dev.mysql.com/doc/refman/5.7/en/set-statement.html

table of contents on the left shows:

```
SQL Statement Syntax
>    Database Administration Statements
      >    SET Syntax
            >    SET Syntax for Variable Assignment
       
```